gadget/gadget.go

```diff
@@ -937,32 +937,49 @@ func IsCompatible(current, new *Info) error {
 	return nil
 }
 
-// LaidOutVolumeFromGadget takes a gadget rootdir and lays out the
-// partitions as specified.
-func LaidOutVolumeFromGadget(gadgetRoot string, model Model) (*LaidOutVolume, error) {
-	info, err := ReadInfo(gadgetRoot, model)
+// LaidOutSystemVolumeFromGadget takes a gadget rootdir and lays out the
+// partitions as specified. It returns one specific volume, which is the volume
+// on which system-* roles/partitions exist, all other volumes are assumed to
+// already be flashed and managed separately at image build/flash time, while
+// the ubuntu-* roles can be manipulated on the returned volume during install
+// mode.
+func LaidOutSystemVolumeFromGadget(gadgetRoot string, model Model) (*LaidOutVolume, error) {
+	// model should never be nil here
+	if model == nil {
+		return nil, fmt.Errorf("internal error: must have model to lay out system volumes from a gadget")
+	}
+	// rely on the basic validation from ReadInfo to ensure that the ubuntu-*
```

Comment:
```suggestion
	// rely on the basic validation from ReadInfo to ensure that the system-*
```

Reply:
followup is https://github.com/snapcore/snapd/pull/9864